openwisp_monitoring/check/tasks.py

```diff
@@ -54,3 +53,21 @@ def auto_create_ping(model, app_label, object_id, created):
     check = Check(name='Ping', check=ping_path, content_type=ct, object_id=object_id)
     check.full_clean()
     check.save()
+
+
+@shared_task
+def find_related_checks(object_id):
+    """
+    Retrieves all related checks to the passed ``device``
+    and calls the ``perform_check`` task from each of them.
+    If no check exists changes the status to ``OK``.
+    """
+    iterator = Check.objects.filter(active=True,
+                                    content_type__model='device',
+                                    object_id=object_id) \
+                            .only('id') \
+                            .values('id')
+    if not list(iterator):
+        DeviceMonitoring.objects.get(device_id=object_id).update_status('ok')
+    for check in iterator:
+        perform_check.delay(check['id'])
```

Comment:
This could have be rewritten as:

```python
device = DeviceData.objects.get(pk=pk)
checks = device.checks.filter(active=True).only('id').values('id')
for check in checks:
    perform_check.delay(check['id'])
# if there's no available check for this device, we'll flag it as OK directly
else:
    device.monitoring.update_status('ok')
```

Anyway, I don't think this is the right place to put this code.
I would rename this as `trigger_device_recovery` and move it to `openwisp_monitoring.device` since most of the changes are being done there.